dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/graph/WorkflowGraphTopologyLogicalVisitor.java

```diff
@@ -143,8 +143,11 @@ private void doVisitationInSubGraph(Set<String> subGraphNodes) {
             }
             for (String successor : successors) {
                 inDegreeMap.put(successor, inDegreeMap.get(successor) - 1);
+                // Add all non-repeating successors to bootstrapTaskCodes
+                if (!bootstrapTaskCodes.contains(successor)) {
+                    bootstrapTaskCodes.add(successor);
+                }
```

Comment:
There will still exist problem, please use a visited table to record the node which has been visted..